subsys/bluetooth/host/conn.c

```diff
@@ -2291,38 +2327,15 @@ struct bt_conn *bt_conn_create_slave_le(const bt_addr_le_t *peer,
 
 	conn = bt_conn_lookup_addr_le(param->id, peer);
 	if (conn) {
-		switch (conn->state) {
-		case BT_CONN_CONNECT_DIR_ADV:
-			/* Handle the case when advertising is stopped with
-			 * bt_le_adv_stop function
-			 */
-			err = bt_le_adv_start_internal(&param_int, NULL, 0,
-						       NULL, 0, peer);
-			if (err && (err != -EALREADY)) {
-				BT_WARN("Directed advertising could not be"
-					" started: %d", err);
-				bt_conn_unref(conn);
-				return NULL;
-			}
-
-		case BT_CONN_CONNECT:
-		case BT_CONN_CONNECTED:
-			return conn;
-		case BT_CONN_DISCONNECTED:
-			BT_WARN("Found valid but disconnected conn object");
-			goto start_adv;
-		default:
-			bt_conn_unref(conn);
-			return NULL;
-		}
+		bt_conn_unref(conn);
```

Comment:
A BT_WARN() might be appropriate here, especially to give hints to out-of-tree apps that might have relied on a successful return previously.

Reply:
Added.

Reply:
Added.